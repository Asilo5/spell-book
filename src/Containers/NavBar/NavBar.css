nav {
    display: flex;
    justify-content: space-around;
    align-items: center;
    text-align: center;
    padding-bottom: 20px;
    background: linear-gradient( to left, #cc9721 22%,#ddab56 25%, #d4bc83 40%, #ccb37c 31%,#af8d25 45%,#ebb954 80%) left bottom transparent no-repeat;
    background-size: 100% 5px;
}

h2 {
    background: linear-gradient(to bottom, #cfc09f 22%,#cc8914 24%, #cfc09f 26%, #cfc09f 27%,#ffecb3 40%,#a36f04 78%); 
    -webkit-background-clip: text;
    -webkit-text-fill-color: transparent;
    color: #fff;
    font-family: 'Playfair Display', serif;
    position: relative;
	text-transform: uppercase;	
	font-size: 3vw;
	margin: 0;
	font-weight: 400;
}

.mascot-img {
    border-radius: 50%;
    padding: 12px;
    margin: 10px;
    height: 14em;
    width: 14em;
    background: linear-gradient( to left, #cc9721 22%,#ddab56 25%, #d4bc83 40%, #ccb37c 31%,#af8d25 45%,#ebb954 80%) left bottom transparent no-repeat;
    transition: all .2s ease-in-out;
}

.mascot-img:hover {
  transform: scale(1.1);
}