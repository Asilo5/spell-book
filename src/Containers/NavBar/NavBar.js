import React from 'react';
import { connect } from 'react-redux';
import './NavBar.css';
import { Link, NavLink } from 'react-router-dom'; 
import { PropTypes } from 'prop-types';

export const NavBar = ({userHouse, nameOfWizard}) => {
    return ( 
        <nav>
            <h2>{nameOfWizard} Potter </h2>
            <Link to='/'>
              {userHouse === 'Gryffindor' && <img className='mascot-img' src='https://animeandthings.com/wp-content/uploads/2018/07/Gryffindor.png' alt='Gryffindor mascot'/> }
              {userHouse === 'Hufflepuff' && <img className='mascot-img' src='http://www.colleenmarielynch.com/Hufflepuff/crest2.gif' alt='Hufflepuff mascot'/>}
              {userHouse === 'Ravenclaw' && <img className='mascot-img' src='https://i.pinimg.com/originals/35/4d/01/354d01e48c6acd596e48f6c4bbb3b7da.jpg' alt='Ravenclaw mascot'/> }
              {userHouse === 'Slytherin' && <img className='mascot-img' src='https://www.pngkey.com/png/detail/226-2269266_slytherin-crest-png-harry-potter-slytherin-logo.png' alt='Slytherin mascot'/> }
            </Link>
            <NavLink to='/favourites'>
              <h2 className='fav-link'>Favourite Spells</h2>
            </NavLink>
        </nav>
    )
}

export const mapStateToProps = ({userHouse, nameOfWizard}) => ({
   userHouse,
   nameOfWizard
})

NavBar.propTypes = {
   userHouse: PropTypes.string,
   nameOfWizard: PropTypes.string
};

export default connect(mapStateToProps)(NavBar);