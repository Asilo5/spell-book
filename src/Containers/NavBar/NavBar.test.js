import React from 'react';
import { shallow } from 'enzyme';
import { NavBar, mapStateToProps } from './NavBar';

describe('NavBar', () => {
    describe('Nav', () => {
      let mockHouse = 'Hufflepuff';
      let mockName = 'Bob Potter';

      let wrapper = shallow(<NavBar 
         userHouse={mockHouse}
         nameOfWizard={mockName}
      />)

      it('should match snapshot with all information passing in correctly', () => {
        expect(wrapper).toMatchSnapshot();
      })
    })

    describe('mapStateToProps', () => {
       it('should return userHouse state', () => {
           const mockState = {
            "userHouse": 'HufflePuff',
            "nameOfWizard": 'Consuelo Weasley'
           };
           
           const expected = {
             "userHouse": 'HufflePuff',
             "nameOfWizard": 'Consuelo Weasley'
           };

           const mappedProps = mapStateToProps(mockState);
           expect(mappedProps).toEqual(expected);
       })
    })
})