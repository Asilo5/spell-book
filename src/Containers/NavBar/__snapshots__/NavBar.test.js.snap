// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar Nav should match snapshot with all information passing in correctly 1`] = `
<nav>
  <h2>
    Bob Potter
     Potter 
  </h2>
  <Link
    to="/"
  />
  <NavLink
    to="/favourites"
  >
    <h2
      className="fav-link"
    >
      Favourite Spells
    </h2>
  </NavLink>
</nav>
`;
